File: BubbleSort.java
// 03-BubbleSort.java

public class BubbleSort {

    // Method to perform Bubble Sort
    public static void bubbleSort(int[] array) {

        // Get array size
        int n = array.length;

        // Outer loop for each pass
        for (int i = 0; i < n - 1; i++) {

            // Inner loop for comparing adjacent elements
            for (int j = 0; j < n - i - 1; j++) {

                // Swap if current element is greater than next element
                if (array[j] > array[j + 1]) {

                    int temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;

                }

            }

        }

    }

    // Method to print array
    public static void printArray(int[] array) {

        for (int number : array) {
            System.out.print(number + " ");
        }

        System.out.println();
    }

    public static void main(String[] args) {

        // Unsorted array
        int[] numbers = {64, 34, 25, 12, 22, 11, 90};

        System.out.println("Before Sorting:");
        printArray(numbers);

        // Call Bubble Sort
        bubbleSort(numbers);

        System.out.println("After Sorting:");
        printArray(numbers);

    }
}

File: BubbleSort.c
#include <stdio.h>

int main() {
    int arr[100];
    int n, i, j, temp;

    // Input array size
    printf("Enter the number of elements: ");
    scanf("%d", &n);

    // Input array elements
    printf("Enter %d elements:\n", n);
    for(i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    // Bubble Sort
    for(i = 0; i < n - 1; i++) {

        for(j = 0; j < n - i - 1; j++) {

            if(arr[j] > arr[j + 1]) {

                temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;

            }

        }

    }

    // Display sorted array
    printf("\nSorted Array (Ascending Order):\n");

    for(i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }

    printf("\n");

    return 0;
}
